nixos/tests/networking.nix

```diff
@@ -102,35 +102,35 @@ let
       };
       testScript = { ... }:
         ''
-          startAll;
+          start_all()
 
-          $client->waitForUnit("network.target");
-          $router->waitForUnit("network-online.target");
+          client.wait_for_unit("network.target")
+          router.wait_for_unit("network-online.target")
 
           # Make sure dhcpcd is not started
-          $client->fail("systemctl status dhcpcd.service");
+          client.fail("systemctl status dhcpcd.service")
 
           # Test vlan 1
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.3");
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.10");
+          client.wait_until_succeeds("ping -c 1 192.168.1.1")
+          client.wait_until_succeeds("ping -c 1 192.168.1.2")
+          client.wait_until_succeeds("ping -c 1 192.168.1.3")
+          client.wait_until_succeeds("ping -c 1 192.168.1.10")
 
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.3");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.10");
+          router.wait_until_succeeds("ping -c 1 192.168.1.1")
+          router.wait_until_succeeds("ping -c 1 192.168.1.2")
+          router.wait_until_succeeds("ping -c 1 192.168.1.3")
+          router.wait_until_succeeds("ping -c 1 192.168.1.10")
 
           # Test vlan 2
-          $client->waitUntilSucceeds("ping -c 1 192.168.2.1");
-          $client->waitUntilSucceeds("ping -c 1 192.168.2.2");
+          client.wait_until_succeeds("ping -c 1 192.168.2.1")
+          client.wait_until_succeeds("ping -c 1 192.168.2.2")
 
-          $router->waitUntilSucceeds("ping -c 1 192.168.2.1");
-          $router->waitUntilSucceeds("ping -c 1 192.168.2.2");
+          router.wait_until_succeeds("ping -c 1 192.168.2.1")
+          router.wait_until_succeeds("ping -c 1 192.168.2.2")
 
           # Test default gateway
-          $router->waitUntilSucceeds("ping -c 1 192.168.3.1");
-          $client->waitUntilSucceeds("ping -c 1 192.168.3.1");
+          router.wait_until_succeeds("ping -c 1 192.168.3.1")
+          client.wait_until_succeeds("ping -c 1 192.168.3.1")
```

Comment:
why not:

```python
vlan1_ips = [  "192.168.1.1", " 192.168.1.2", " 192.168.1.3",  "192.168.1.10" ]
vlan2_ips = [  "192.168.2.1", " 192.168.2.2" ]
gateway_ip = "192.168.3.1"

def ping_until_succeeds(machine, addr):
    machine.wait_until_succeeds(f"ping -c 1 {addr}")

with subtest("Test VLAN 1"):
    for machine in client, router:
        for ip in vlan1_ips:
            ping_until_succeeds(machine, ip)

with subtest("Test VLAN 2"):
     for machine in client, router:
         for ip in vlan2_ips:
            ping_until_succeeds(machine, ip)

with subtest("Test default gateway"):
    ping_until_succeeds(router, gateway_ip)
    ping_until_succeeds(client, gateway_ip)
```

Reply:
https://github.com/NixOS/nixpkgs/pull/75721#discussion_r358139172